src/Domain/FunctionsOfTime/Tags.hpp

```diff
@@ -19,11 +21,36 @@ template <size_t VolumeDim>
 class DomainCreator;
 /// \endcond
 
-namespace domain {
-namespace Tags {
+namespace domain::Tags {
+/// The functions of time obtained from a domain creator, exceipt if
+/// `OverrideCubicFunctionsOfTime` is true, then the cubic piecewise polynomial
+/// functions overriden using `read_spec_third_order_piecewise_polynomial()`
+template <size_t Dim, bool OverrideCubicFunctionsOfTime = false>
+struct InitialFunctionsOfTime {};
+
 /// The functions of time obtained from a domain creator
 template <size_t Dim>
-struct InitialFunctionsOfTime : db::SimpleTag {
+struct InitialFunctionsOfTime<Dim, true> : db::SimpleTag {
+  using type = std::unordered_map<
+      std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>;
+  using option_tags =
+      tmpl::list<domain::OptionTags::DomainCreator<Dim>,
+                 domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile,
+                 domain::FunctionsOfTime::OptionTags::FunctionOfTimeNameMap>;
+
+  static constexpr bool pass_metavariables = false;
+  static type create_from_options(
+      const std::unique_ptr<::DomainCreator<Dim>>& domain_creator,
+      const std::optional<std::string>& function_of_time_file,
+      const std::optional<std::map<std::string, std::string>>&
+          function_of_time_name_map) noexcept;
+};
+
+/// The functions of time obtained from a domain creator, with
+/// cubic piecewise polynomial functions overriden using
+// `read_spec_third_order_piecewise_polynomial()`
```

Comment:
Swap this comment with the one of the template specialization above.

Reply:
🤦‍♂️ Thanks for catching this!